cmake/scripts/linux/ArchSetup.cmake

```diff
@@ -41,6 +41,6 @@ if(CMAKE_BUILD_TYPE STREQUAL Coverage)
   set(COVERAGE_EXCLUDES */test/* lib/* */lib/*)
 endif()
 
-if(ENABLE_MIR)
+if(ENABLE_MIR OR ENABLE_GBM)
```

Comment:
that is rather hacky for more than one reason. you print a message with "Mir" when not building for Mir. but the real problem is the inverted logic of the build system. it arbitrarily enables vdpau and the various platforms need to disable it. vdpau does only work with X11 and should only be enabled for this platform.